docs/book/src/topics/azure-cni-v1.md

```diff
@@ -0,0 +1,138 @@
+      .
+    ```
+
+- `kind: KubeadmControlPlane` of control plane nodes
+  - add `max-pods: "110"` to `spec.kubeadmConfigSpec.initConfiguration.nodeRegistration.kubeletExtraArgs`.
+  - add `max-pods: "110"` to `spec.kubeadmConfigSpec.joinConfiguration.nodeRegistration.kubeletExtraArgs`.
+
+  - ```yaml
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: KubeadmControlPlane
+    metadata:
+      name: ${CLUSTER_NAME}-control-plane
+      namespace: default
+    spec:
+      kubeadmConfigSpec:
+        .
+        .
+        initConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              max-pods: "110"
+              .
+              .
+        joinConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              max-pods: "110"
+              .
+              .
+    ```
+
+- `kind: AzureMachineTemplate` of control-plane
+  - Add `networkInterfaces` to controlplane's `AzureMachineTemplate`
+
+  - ```yaml
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: AzureMachineTemplate
+    metadata:
+      name: ${CLUSTER_NAME}-control-plane
+      namespace: default
+    spec:
+      template:
+        spec:
+          .
+          .
+          networkInterfaces:
+          - privateIPConfigs: 110 # maximum private IPs to be allocated in this node
+            subnetName: control-plane-subnet
+          .
+          .
+    ```
+
+- `kind: AzureMachineTemplate` of worker node
+  - Add `networkInterfaces` to worker node's `AzureMachineTemplate`
+
+  - ```yaml
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: AzureMachineTemplate
+    metadata:
+      name: ${CLUSTER_NAME}-md-0
+      namespace: default
+    spec:
+      template:
+        spec:
+          networkInterfaces:
+          - privateIPConfigs: 110
+            subnetName: node-subnet
+          .
+          .
+    ```
+
+- `kind: KubeadmControlPlane` of worker nodes
+  - add `max-pods: "110"` to `spec.template.spec.joinConfiguration.nodeRegistration.kubeletExtraArgs`.
+
+  - ```yaml
+    apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+    kind: KubeadmConfigTemplate
+    metadata:
+      name: ${CLUSTER_NAME}-md-0
+      namespace: default
+    spec:
+      template:
+        spec:
+          .
+          .
+          joinConfiguration:
+            nodeRegistration:
+              kubeletExtraArgs:
+                max-pods: "110"
+                .
+                .
+    ```
+
+## Limitations
+
+- We can only configure one subnet per control-plane node. Refer [CAPZ#3506](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/3506)
+
+- We can only configure one Network Interface per worker node. Refer[Azure-container-networking#3611](https://github.com/Azure/azure-container-networking/issues/1945)
+
+<!-- TODO: Do we specify the number of IPs per nodes depending on the VM size? because Refer https://learn.microsoft.com/en-us/azure/aks/configure-azure-cni#maximum-pods-per-node -->
```

Comment:
are these still TODO?

Reply:
Not really. Updated the docs and removed these TODOs.